specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json

```diff
@@ -540,6 +641,64 @@
         }
       }
     },
+    "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}/backupProtectionIntents": {
+      "get": {
+        "tags": [
+          "BackupProtectionIntent"
+        ],
+        "description": "Provides a pageable list of all intents that are present within a vault.",
+        "operationId": "BackupProtectionIntent_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/VaultName"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "description": "OData filter options.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$skipToken",
+            "in": "query",
+            "description": "skipToken Filter.",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ProtectionIntentResourceList"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-odata": "#/definitions/ProtectionIntentQueryObject",
```

Comment:
what does this x-ms-odata do, should it be modeled differently @anuchandy is this odata query parameter needed in the URL?

Reply:
This is the query object. Yes, it is the odata query parameter.

Reply:
`ProtectionIntentQueryObject` type describes valid query parameter names and their values that can be used in OData filter condition.

https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-odata